### DISCUSSION TITLE
Getting current uftrace to users: Debian (and Linuxbrew)

### DISCUSSION THREAD

**Discussion by bernhardkaindl** [namhyung/uftrace#1327] [Show and tell]
To build uftrace using LinuxBrew/Homebrew on Linux, do this:

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`
and follow the instructions shown there. Then, you run simply:

`brew install -s uftrace`

This downloads all dependencies of uftrace (already pre-built) and builds uftrace from source with all features (luajit support is coming soon).

If you want the latest uftrace from the git HEAD, you can build it from the master branch using:

`brew install -s --HEAD uftrace`

The full set of uftrace manual pages are provided as well.

The best thing is: If you don't have to have `sudo` privileges on the machine, it simply installs into your home directory, if you don't.

In case you have `sudo` privileges, it will install below `/home/linuxbrew/.linuxbrew`, so it does not pollute your `/usr/local`.

In any case, any `Linuxbrew`-built tools are only in your `PATH` as long as you source the output of `brew shellenv` as described into the installation instructions. If you remove the corresponding line from you `~/.profile`, or don't even add it, you can call the installed tools like `uftrace` using the direct path or by a symlink to it.

The "formula" to brew `uftrace` can be maintained by anyone who has a github account, just create a fork of homebrew-core on github, commit your changes to a new branch and create a pull request. Here is my currently-active pull request which adds support for luajit: https://github.com/Homebrew/homebrew-core/pull/84316

In case you have questions, just write a comment here in the discussion!

**Comment by alexmyczko** [comment]
debian users can be happy, it's updated in sid, and a bug has been filed to remove armhf binary so it can migrate to testing soon

**Reply by alexmyczko** [reply]
if someone needs it in older debian, please create a backport request on debian bug tracking system
